presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java

```diff
@@ -875,6 +878,15 @@ public void testPartitionSchemaMismatch()
         getAllSplits(splitManager.getPartitionSplits(table, partitionResult.getPartitions()));
     }
 
+    public void testPartitionSchemaNonCanonical()
+            throws Exception
+    {
+        ConnectorTableHandle table = getTableHandle(tablePartitionSchemaChangeNonCanonical);
+        ConnectorPartitionResult partitionResult = splitManager.getPartitions(table, TupleDomain.<ConnectorColumnHandle>all());
```

Comment:
I think the type hint is not required here (IDEA seems not to be updated in this case for Java 8's improved type inference)